src/utils/hooks/useSearchPublications.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useDispatch } from 'react-redux';
+import { search } from '../../api/publicationApi';
+import { setNotification } from '../../redux/actions/notificationActions';
+import { NotificationVariant } from '../../types/notification.types';
+import { SearchResult } from '../../types/search.types';
+import useCancelToken from './useCancelToken';
+
+const useSearchPublications = (
+  searchTerm: string | null,
+  numberOfResults?: number,
+  searchAfter?: string
+): [SearchResult[] | [], boolean] => {
+  const dispatch = useDispatch();
+  const { t } = useTranslation('feedback');
+  const cancelToken = useCancelToken();
+  const [isLoading, setIsLoading] = useState(true);
+  const [searchResults, setSearchResults] = useState<SearchResult[]>([]);
+
+  useEffect(() => {
+    const searchPublications = async () => {
```

Comment:
denne burde kanskje flyttes utenfor `useEffect` og  også returneres, slik at brukeren kan gjøre flere søk på rappen uten at denne må mountes på nytt?

Reply:
skjønner ikke helt hva du mener her? mener du at `searchPublications`-funksjonen skal returneres fra denne hooken? hva er poenget med det? søket vil jo bli gjennomført når inputen (searchTerm) endres?

Reply:
og vi vil vel at denne funksjonen skal mountes på nytt hver gang pga annen input? vil vel bli akkurat samme performance om den er definert utenfor useEffect eller med useCallback med searchTerm som dependency?

Reply:
ah, jeg trodde det var en søkeknapp som trigga søket direkte, men gikk opp for meg nå at alt går gjennom URLen.
apropos: query i URL burde være startverdi i searchbar hvis man kommer dit via link elns

Reply:
vet ikke helt om jeg skjønte din apropos. Hvis man får en link ala `http://localhost:3000/search?query=asdf`, og limer inn, så får man jo opp samme resultat?

Reply:
eller mente du at det burde stå `asdf` i søkeboksen?

Reply:
fikset sånn at det står `asdf` i søkeboksen når man limer inn lenke nå

Reply:
ja, for man vil jo få vist søketreffene for `asdf` under